### ISSUE TITLE
leanproject up access is denied

### ISSUE THREAD

**Issue by kckennylau** [leanprover-community/mathlib-tools#46]
Error:
```
PermissionError: [WinError 5] Access is denied: 'C:\\some-repo\\_target\\deps\\mathlib\\.git\\objects\\pack\\pack-36f591561703d876c8ad32e4cc46887e2a0bf6d2.idx'
```

Environment: VSCode Terminal on Windows 10, in some repo that uses mathlib.

Reproduction:
1. `leanproject up`
2. `leanproject up`

Debug:
```text
Traceback (most recent call last):
  File "c:\python-3.6.4\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\python-3.6.4\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Python-3.6.4\Scripts\leanproject.exe\__main__.py", line 7, in <module>
  File "c:\python-3.6.4\lib\site-packages\mathlibtools\leanproject.py", line 256, in safe_cli
    handle_exception(err, str(err))
  File "c:\python-3.6.4\lib\site-packages\mathlibtools\leanproject.py", line 51, in handle_exception
    raise exc
  File "c:\python-3.6.4\lib\site-packages\mathlibtools\leanproject.py", line 254, in safe_cli
    cli()
  File "c:\python-3.6.4\lib\site-packages\click\core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "c:\python-3.6.4\lib\site-packages\click\core.py", line 782, in main
    rv = self.invoke(ctx)
  File "c:\python-3.6.4\lib\site-packages\click\core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "c:\python-3.6.4\lib\site-packages\click\core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\python-3.6.4\lib\site-packages\click\core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "c:\python-3.6.4\lib\site-packages\mathlibtools\leanproject.py", line 92, in upgrade_mathlib
    proj().upgrade_mathlib()
  File "c:\python-3.6.4\lib\site-packages\mathlibtools\lib.py", line 468, in upgrade_mathlib
    shutil.rmtree(str(self.mathlib_folder))
  File "c:\python-3.6.4\lib\shutil.py", line 494, in rmtree
    return _rmtree_unsafe(path, onerror)
  File "c:\python-3.6.4\lib\shutil.py", line 384, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "c:\python-3.6.4\lib\shutil.py", line 384, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "c:\python-3.6.4\lib\shutil.py", line 384, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "c:\python-3.6.4\lib\shutil.py", line 389, in _rmtree_unsafe
    onerror(os.unlink, fullname, sys.exc_info())
  File "c:\python-3.6.4\lib\shutil.py", line 387, in _rmtree_unsafe
    os.unlink(fullname)
PermissionError: [WinError 5] Access is denied: 'C:\\some-repo\\_target\\deps\\mathlib\\.git\\objects\\pack\\pack-36f591561703d876c8ad32e4cc46887e2a0bf6d2.idx'
```

For example: https://github.com/kckennylau/test

**Comment by jcommelin** [comment]
What happens if you try another terminal? Is this a `leanproject` issue, or something that's specific to the VScode terminal?

**Comment by jcommelin** [comment]
For the record, I cannot reproduce this on Linux. `leanproject up` feels a bit slow when I execute it in your project, but it succeeds, and the second time it succeeds again.

**Comment by kckennylau** [comment]
Reproduced in git bash.

**Comment by kbuzzard** [comment]
More info at https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/leanproject.20access.20denied

**Comment by kckennylau** [comment]
workaround seems to be to execute `leanproject up` inside `_target/deps/mathlib`

**Comment by PatrickMassot** [comment]
Fixed in 1918d2c55386aad5be81e0efb82896978c2e0121
